src/Domain/Handler/SubscriptionRefundedEventHandler.ts

```diff
@@ -46,15 +48,22 @@ implements DomainEventHandlerInterface
   }
 
   private async updateUserRole(user: User): Promise<void> {
-    const role = await this.roleRepository.findOneByName(RoleName.CoreUser)
+    const currentRoleName = (await user.roles)[0].name as RoleName
+    const newRoleName = RoleName.CoreUser
+    const newRole = await this.roleRepository.findOneByName(newRoleName)
 
-    if (role === undefined) {
+    if (newRole === undefined) {
       this.logger.warn(`Could not find role for role name: ${RoleName.CoreUser}`)
       return
     }
 
-    user.roles = Promise.resolve([role])
+    user.roles = Promise.resolve([newRole])
```

Comment:
Hmm let's discuss this :) 

I think the role should be appended instead of overwritten. For example we might at some point create some custom roles for custom users - the general idea is that a user can have multiple roles instead of one. Wdyt?

Reply:
Makes sense 🙂 will do!

Reply:
we have to remove the role upon subscription cancelling/refunding, hmm and also expiring.

Reply:
With that in mind, maybe the subscription events could include a "formSubscriptionPlan" and a "toSubscriptionPlan" field so that we know which role to append and which one to remove? Wdyt? Not sure about expiring though, perhaps we could have a scheduled job?

Reply:
Yes, that makes more sense 👍  - since a user can be only on one subscription plan. 

Hmm that would also make sense to change the name of the even from `USER_ROLE_CHANGED` to `USER_SUBSCRIPTION_PLAN_CHANGED` right?

As for expiring I think there already is a job for that in payments service but it doesn't emit an event. I think we should add this as a separate task in Asana.

Reply:
I think we could leave the `USER_ROLE_CHANGED` event as is so we can use it to ping clients about a new role if we had custom roles at some point, what do you think? The "fromPlan" and "toPlan" fields would be included in the other events

Reply:
To keep track of things, this is the only change pending for this PR which is dependent on this one https://github.com/standardnotes/snjs/pull/368